tests/unit/test_artifactory_path.py

```diff
@@ -20,17 +22,40 @@
 
 class UtilTest(unittest.TestCase):
     def test_matrix_encode(self):
-        params = {"foo": "bar", "qux": "asdf"}
+        with mock.patch("urllib.parse.quote", new=mock.Mock(wraps=quote_original)) as q:
 
-        s = artifactory.encode_matrix_parameters(params)
+            params = {"foo": "bar", "qux": "asdf"}
 
-        self.assertEqual(s, "foo=bar;qux=asdf")
+            s = artifactory.encode_matrix_parameters(params, quote_parameters=False)
 
-        params = {"baz": ["bar", "quux"], "foo": "asdf"}
+            self.assertEqual(s, "foo=bar;qux=asdf")
+            q.assert_not_called()
 
-        s = artifactory.encode_matrix_parameters(params)
+            params = {"baz": ["bar", "quux"], "foo": "asdf"}
 
-        self.assertEqual(s, "baz=bar;baz=quux;foo=asdf")
+            s = artifactory.encode_matrix_parameters(params, quote_parameters=False)
+
+            self.assertEqual(s, "baz=bar;baz=quux;foo=asdf")
+            q.assert_not_called()
+
+            # Test with quoting
+            params = {"b?az": ["b%ar", "quux"], "foo?%0": "a/s&d%f?"}
+
+            s = artifactory.encode_matrix_parameters(params, quote_parameters=True)
+
+            expected_calls = []
+            for k, v in params.items():
```

Comment:
Looks a bit complicated for tests. Can we make it simple assertions?

Reply:
This is checking that the exact calls we expect are all happening. Could you give an example of how to achieve that with simple assertions?

Or do you prefer not to check those calls?

Reply:
Instead of building `expected_calls` we can explicecly build it
```python
expected_calls = [
    mock.call('key', 'value'),
    mock.call('key2', 'value2'),
    ...
]
```

But here - yes, let's remove mocks and assert against the final step - api call

Reply:
I have removed the wrapped calls completely in https://github.com/devopshq/artifactory/pull/409/commits/2f1551c439924ecd0537ea01d806f9686607146c as requested.